src/pages/Group/GroupSettings.tsx

```diff
@@ -0,0 +1,143 @@
+import { useEffect, useState } from "react";
+import toast from "react-hot-toast";
+import { useMutation, useQueryClient } from "react-query";
+import ToastMessage from "../../components/ToastMessage";
+import Column from "../../components/layout/Column";
+import Row from "../../components/layout/Row";
+import { QueryKeys } from "../../constants";
+import { useApi } from "../../contexts/ApiProvider";
+import { Group } from "../../data/models/Group";
+
+import { useNavigate } from "react-router-dom";
+import LeftArrowIcon from "../../assets/arrow_left.svg";
+import Button from "../../components/Button";
+import Spinner from "../../components/Spinner";
+import TextField from "../../components/TextField";
+import ClosableModal from "../../components/modals/ClosableModal";
+import { useUser } from "../../contexts/UserProvider";
+
+export default function GroupSettings({
+  group,
+  setShowGroupSettings,
+}: {
+  group: Group;
+  setShowGroupSettings: React.Dispatch<React.SetStateAction<boolean>>;
+}) {
+  const { user } = useUser();
+  const navigate = useNavigate();
+
+  // group settings:
+  const [groupNameText, setGroupNameText] = useState("");
+  const [leaveError, setLeaveError] = useState("");
+  const [changeGroupNameError, setChangeGroupNameError] = useState("");
+  const { api } = useApi();
+  const queryClient = useQueryClient();
+
+  // leave modal
+  const confirmLeaveId = "confirm-leave";
+
+  const { mutate: saveGroupName, isLoading: putGroupNameLoading } =
```

Comment:
something interesting I observed when changing group name, the order of the groups on the user groups page changes. Do you think you could just sort them by group id so the order stays the same?